x/proof/keeper/msg_server_create_claim.go

```diff
@@ -52,6 +42,20 @@ func (k msgServer) CreateClaim(
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	// Defer telemetry calls so that they reference the final values the relevant variables.
+	defer func() {
```

Comment:
Can you move this into a separate function so you just defer that?

It'll help separate telemetry logic from business logic.